src/main/java/com/internship/backend/service/ReservationService.java

```diff
@@ -39,34 +38,58 @@ public List<Reservation> getAllReservations(){
         return reservationRepository.findAll();
     }
 
-    public Reservation addReservation(Reservation reservation) throws ReservationAlreadyExists {
+    public Reservation addReservation(Reservation reservation) throws ReservationAlreadyExists, InvalidDateException {
 
-        if (isValidReservation((reservation)))
-        {
+
+        addReservationValidation(reservation);
+
+        if (isValidReservation((reservation))) {
             return reservationRepository.save(reservation);
         }
         else{
             throw new ReservationAlreadyExists("Reservation intersects another one");
         }
     }
 
-    public boolean isValidReservation(Reservation reservation){
+    public void addReservationValidation(Reservation reservation) throws InvalidDateException {
+
+        if(!reservation.getStartTime().isWithinNext24Hours()){
+            throw new InvalidDateException("Reservations can only be made for the current date or within the next 24 hours");
+        }
+
+        if (reservation.getStartTime().getMonth() != reservation.getEndTime().getMonth()) {
+            throw new InvalidDateException("The start date and end date must be within the same month");
+        }
+
+        if(reservation.getStartTime().getDay() != reservation.getEndTime().getDay()){
+            throw new InvalidDateException("The start date and end date must be within the same day");
+        }
+
+        if(reservation.getStartTime().getYear() != reservation.getEndTime().getYear()){
+            throw new InvalidDateException("The start date and end date must be within the same year");
+        }
+
+        if(reservation.getEndTime().getHour() - reservation.getStartTime().getHour() != 2){
+            throw new InvalidDateException("The reservation duration must be exactly 2 hours");
+        }
```

Comment:
Could this method be incapsulated in a class, having a field named reservation and a public method called validate() ? Also, the error messages could be defined as constants.